code/game/objects/items/stacks/hot_ice.dm

```diff
@@ -0,0 +1,26 @@
+/obj/item/stack/hot_ice
+	name = "hot ice"
+	icon_state = "hot-ice"
+	item_state = "hot-ice"
+	singular_name = "hot ice"
+	icon = 'icons/obj/stack_objects.dmi'
+	custom_materials = list(/datum/material/hot_ice=10000)
+	grind_results = list(/datum/reagent/toxin/plasma = 200)
+	material_type = /datum/material/hot_ice
+
+/obj/item/stack/hot_ice/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins licking \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return FIRELOSS//dont you kids know that stuff is toxic?
+
+/obj/item/stack/hot_ice/attackby(obj/item/W as obj, mob/user as mob, params)
```

Comment:
this should be a component for the material

Reply:
what you mean?

Reply:
he's saying the effects of this should be on the actual material datum

Reply:
ah i just copied the one in the plasma sheet in /obj/item/stack/sheet/mineral/plasma
its the same, and the plasma doesnt have anything like that in its basemats.dm so i didnt change it. 
basically what i did here is make into basemats.dm a copy of glass and plasma materials to create the hot_ice material, and then i took the plasma sheets from mineral.dm and made a new file called hot_ice.dm and changed it to make it work with that ice

Reply:
Yup sadly datum materials are relatively new but because of the amount of code that affects materials things like sheets having specific code like that is still in the works of being fixed. But we would prefer future things to already have it fixed